src/compiler/HipsyclClangPlugin.cpp

```diff
@@ -40,38 +45,77 @@ namespace hipsycl {
 namespace compiler {
 // Register and activate passes
 
-static clang::FrontendPluginRegistry::Add<hipsycl::compiler::FrontendASTAction>
-    HipsyclFrontendPlugin{"hipsycl_frontend", "enable hipSYCL frontend action"};
+static clang::FrontendPluginRegistry::Add<hipsycl::compiler::FrontendASTAction> HipsyclFrontendPlugin{
+    "hipsycl_frontend", "enable hipSYCL frontend action"};
 
-static void registerGlobalsPruningPass(const llvm::PassManagerBuilder &,
-                                       llvm::legacy::PassManagerBase &PM) {
+static void registerGlobalsPruningPass(const llvm::PassManagerBuilder &, llvm::legacy::PassManagerBase &PM) {
   PM.add(new GlobalsPruningPassLegacy{});
 }
 
 static llvm::RegisterStandardPasses
     RegisterGlobalsPruningPassOptLevel0(llvm::PassManagerBuilder::EP_EnabledOnOptLevel0,
-                                        registerGlobalsPruningPass);
+                                           registerGlobalsPruningPass);
 
 static llvm::RegisterStandardPasses
     RegisterGlobalsPruningPassOptimizerLast(llvm::PassManagerBuilder::EP_OptimizerLast,
-                                            registerGlobalsPruningPass);
+                                               registerGlobalsPruningPass);
+
+#if defined(HIPSYCL_USE_ACCELERATED_CPU) && !defined(_WIN32)
+static llvm::RegisterPass<SplitterAnnotationAnalysisLegacy>
+    splitterAnnotationReg("splitter-annot-ana", "hipSYCL splitter annotation analysis pass",
+                          true /* Only looks at CFG */, true /* Analysis Pass */);
```

Comment:
In PR #683, we relay information to the clang plugin using macros, i.e. `syclcc` will define a macro, and the clang plugin checks the existence of the macro as compiler option to adjust its behavior. Would it make sense to have a similar mechanism with the cbs plugin, so that we can disable support even though the clang plugin has been compiled with support for it?

Reply:
Alright - I think it's not super urgent anyway because, as you point out, it shouldn't make a functional difference. It might be useful though if CBS triggers issues with future clang versions etc.